File: src/google/arrays/container_with_most_water.cpp
#include<bits/stdc++.h>

using namespace std;

int maxArea(vector<int>& height) {
	int l=0;
	int r=height.size()-1;
	int max_area = 0;
	while(l<r) {
		if(height[l] < height[r]) {
			max_area = max(max_area, height[l]*(r-l));
			l++;
		} else {
			max_area = max(max_area, height[r]*(r-l));
			r--;
		}
	}
	return max_area;
}

int main() {
	vector<int> a = {4,3,2,1,4};
	cout << maxArea(a);
}

File: src/google/getwellpreparedques/arraysandstrings/ContainerWithMostWater.java
package google.getwellpreparedques.arraysandstrings;

import java.util.*;

public class ContainerWithMostWater {
	public static int maxArea(int[] height) {
        int s = 0;
        int e = height.length - 1;
        int resp = 0;
        while(s<e) {
        	if(height[s] < height[e]) {
        		resp = Math.max(resp, (e-s)*height[s]);
        		s++;
        	} else {
        		resp = Math.max(resp, (e-s)*height[e]);
        		e--;
        	}
        }
        return resp;
    }

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int[] height = new int[n];
		for(int i=0; i<n; i++) {
			height[i] = sc.nextInt();
		}
		System.out.println(maxArea(height));
	}
}